pyo3-stub-gen/src/util.rs

```diff
@@ -0,0 +1,114 @@
+use pyo3::{prelude::*, types::*};
+
+pub fn fmt_py_obj(any: &Bound<'_, PyAny>) -> String {
+    fn all_builtin_types(any: &Bound<'_, PyAny>) -> bool {
+        if any.is_instance_of::<PyString>()
+            || any.is_instance_of::<PyBool>()
+            || any.is_instance_of::<PyInt>()
+            || any.is_instance_of::<PyFloat>()
+            || any.is_none()
+        {
+            return true;
+        }
+        if any.is_instance_of::<PyDict>() {
+            return any
+                .downcast::<PyDict>()
+                .map(|dict| {
+                    dict.into_iter()
+                        .all(|(k, v)| all_builtin_types(&k) && all_builtin_types(&v))
+                })
+                .unwrap_or(false);
+        }
+        if any.is_instance_of::<PyList>() {
+            return any
+                .downcast::<PyList>()
+                .map(|list| list.into_iter().all(|v| all_builtin_types(&v)))
+                .unwrap_or(false);
+        }
+        if any.is_instance_of::<PyTuple>() {
+            return any
+                .downcast::<PyTuple>()
+                .map(|list| list.into_iter().all(|v| all_builtin_types(&v)))
+                .unwrap_or(false);
+        }
+        false
+    }
+    if all_builtin_types(any) {
+        // `to_string` for `PyString` will lose quotes
+        if any.is_instance_of::<PyString>() {
+            if let Ok(s) = any.extract::<String>() {
+                return format!("'{s}'");
+            }
+        } else {
+            return any.to_string();
+        }
+    }
+    "...".to_owned()
```

Comment:
Ah, I am not aware this behaviour. And this formatting seems generate invalid output when it is fed wiith string like `"foo'bar"` or `"buz\\"`.
It seems that python provides [`repr` function to convert into evalable types (for builtin values)](https://docs.python.org/3.10/library/functions.html#repr) and PyO3 provides `Bound::repr` counterpart. So I think this is enough:

```suggestion
    if all_builtin_types(any) {
        any.repr().map_or("...".to_string(), |a| a.to_string())
    } else {
        "...".to_string()
    }
```